src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    /// Utitlies class to manipulate MsBuild projects.
+    /// </summary>
+    internal abstract class MsBuildUtilities
+    {
+        /// <summary>
+        /// Gets a project property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Requested project property. Null if the property is not present.</returns>
+        public static ProjectPropertyElement GetProperty(ProjectRootElement project, string propertyName)
+        {
+            Requires.NotNull(project, "project");
+
+            return project.Properties
+                .Where(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
```

Comment:
That is a good question, on the unevaluated project the last one doesn't necessarily win as it could be inside a conditioned statement, e.g.

`<PropertyGroup Condition="Something">
 <MyProperty>PropertyValue1</MyProperty>
</PropertyGroup>
<PropertyGroup Condition="SomethingElse">
 <MyProperty>PropertyValue2</MyProperty>
</PropertyGroup>`

In this case there will be two properties in the MsBuild object with the name "MyProperty" and without comparing the condition you wouldn't know which one you want. In the case of Configurations and Platforms although you could have them conditioned the current assumption (and how they are getting written when changed via the config manager) is that they are not conditioned and in that case if we somehow end up with multiples you are correct the last one should win. Given that it is a corner case though for perf I would prefer stopping the iteration at the first entry (but I could be convinced otherwise).